### PR TITLE
Typescript 2.6

### PR DESCRIPTION
**Pull request by bryanforbes**
**Type:** enhancement

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Relates to dojo/meta#210

### CODE REVIEW

File: src/Symbol.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ if (!has('es6-symbol')) {
 	};
 
 	const defineProperties = Object.defineProperties;
-	const defineProperty: (o: any, p: string | symbol, attributes: PropertyDescriptor & ThisType<any>) => any = Object.defineProperty;
+	const defineProperty: (o: any, p: string | symbol, attributes: PropertyDescriptor & ThisType<any>) => any = <any> Object.defineProperty;
```
Comment (agubler) [inline src/Symbol.ts]: `as any`?

File: src/support/has.ts [hunk 2]
```diff
@@ -132,7 +132,9 @@ add('es6-string', () => {
 
 add('es6-string-raw', () => {
 	function getCallSite(callSite: TemplateStringsArray, ...substitutions: any[]) {
-		return callSite;
+		const result = [...callSite];
```
Comment (agubler) [inline src/support/has.ts]: nit: code style: `[ ...callSite ]`

File: src/support/has.ts [hunk 3]
```diff
@@ -132,7 +132,9 @@ add('es6-string', () => {
...
+		(<any> result).raw = callSite.raw;
```
Comment (agubler) [inline src/support/has.ts]: nit: `as any`

File: src/support/util.ts [hunk 4]
```diff
@@ -22,8 +22,13 @@ export function getValueDescriptor<T>(value: T, enumerable: boolean = false, wri
  *
  * @param nativeFunction The source function to be wrapped
  */
-export function wrapNative<T, U, R>(nativeFunction: (...args: U[]) => R): (target: T, ...args: U[]) => R {
-	return function (target: T, ...args: U[]): R {
+export function wrapNative<T, U, R>(nativeFunction: (arg1: U) => R): (target: T, arg1: U) => R;
```
Comment (agubler) [inline src/support/util.ts:25]: Is this a change required for the TS2.6 upgrade?

**Review by agubler — COMMENTED** [review]
a few comments :)

File: src/support/util.ts (hunk 4 above)
Comment (bryanforbes) [reply src/support/util.ts:25]: Yes. If the old signature is left as-is, 10 errors occur. As an example, in `src/array.ts` the following error occurs:

```
src/array.ts(125,2): error TS2322: Type '(target: ArrayLike<T>, ...args: number[]) => ArrayLike<T>' is not assignable to type '<T>(target: ArrayLike<T>, offset: number, start: number, end?: number | undefined) => ArrayLike<T>'.
  Types of parameters 'args' and 'end' are incompatible.
    Type 'number | undefined' is not assignable to type 'number'.
      Type 'undefined' is not assignable to type 'number'.
```

**Review by tomdye — APPROVED** [review]

**Pull request merged**